gbus/saga/glue.go

```diff
@@ -148,11 +152,17 @@ func (imsm *Glue) SagaHandler(invocation gbus.Invocation, message *gbus.BusMessa
 			1) If Def does not have handlers for the message type then log a warning (as this should not happen) and return
 			2) Else if the message is a startup message then create new instance of a saga, invoke startup handler and mark as started
 				2.1) If new instance requests timeouts then reuqest a timeout
-			3) Else if message is destinated for a specific saga instance (reply messages) then find that saga by id and invoke it
+			3) Else if message is destinated for a specific saga instance (reply messages / saga is of type SagaIDProvider ) then find that saga by id and invoke it
 			4) Else if message is not an event drop it (cmd messages should have 1 specific target)
 			5) Else iterate over all instances and invoke the needed handler
 		*/
 		startNew := def.shouldStartNewSaga(message)
+		if message.SagaCorrelationID == "" {
+			if sgen, ok := def.newInstance().UnderlyingInstance.(gbus.SagaIDProvider); ok {
```

Comment:
I would rather not create an instance for every message needing correlation only to get a hand on the gbus.SagaIDProvider if implemented.
Ideally, we should do this only once during the RegisterSaga call and have it placed on the Def for future use.

Reply:
Please add documentation and commentary